src/find/mod.rs

```diff
@@ -0,0 +1,476 @@
+mod matchers;
+
+use std::cell::RefCell;
+use std::error::Error;
+use std::io::{Write, stderr, stdout};
+use std::rc::Rc;
+use std::time::SystemTime;
+use walkdir::WalkDir;
+use walkdir::WalkDirIterator;
+
+pub struct Config {
```

Comment:
I could, but I'm struggling to see the benefit. max_depth absolutely should not default to 0. I could change 
```
impl Config {
    fn new() -> Config {
```
to 
```
impl Default for Config {
    fn default() -> Config {
```

But from what I can see I'd then have to change
`let config = Config::new()`
to
`let config : Config = Default::default()`
Which is more to type and requires `Default` to be `use`d. Happy to be persuaded if you can show me how it would be useful (I'm still learning the language after all)